go/vt/vtctl/grpcvtctldserver/server.go

```diff
@@ -2064,7 +2064,7 @@ func (s *VtctldServer) RefreshStateByShard(ctx context.Context, req *vtctldatapb
 		return nil, fmt.Errorf("Failed to get shard %s/%s/: %w", req.Keyspace, req.Shard, err)
 	}
 
-	isPartial, err := topotools.RefreshTabletsByShard(ctx, s.ts, s.tmc, si, req.Cells, logutil.NewCallbackLogger(func(e *logutilpb.Event) {
+	isPartial, _, err := topotools.RefreshTabletsByShard(ctx, s.ts, s.tmc, si, req.Cells, logutil.NewCallbackLogger(func(e *logutilpb.Event) {
```

Comment:
I don't think it makes sense to change the partial handling here, other than we could add the reasons for partial to the response message. I'm going to try and do this. At least then callers can take action and know the causes for it.

Reply:
Done: https://github.com/vitessio/vitess/pull/10058/commits/96b96bbf44126d57e5efeab10c1955a83762f04e

Reply:
👍